src/azure-cli/azure/cli/command_modules/iot/tests/latest/recording_processors.py

```diff
@@ -32,6 +32,9 @@ def _replace_string_keys(self, val):
         if 'secondaryKey' in val:
             val = re.sub(r'"secondaryKey":( ?)"([^"]+)"', r'"secondaryKey":"{}"'
                          .format(MOCK_KEY), val, flags=re.IGNORECASE)
+        if any(['SharedAccessKey=' in val, 'sharedaccesskey=' in val]):
+            val = re.sub(r'[S|s]hared[A|a]ccess[K|k]ey=([^\*].+=)', 'SharedAccessKey={}'
+                         .format(MOCK_KEY), val, flags=re.IGNORECASE)
```

Comment:
Could you add some comments or samples to explain the regular expression to increase it's readability?

Reply:
Sure, though these are just for removing SAS keys from our recorded tests.
The typical connection string response is `SharedAccessKey=***` but one of our calls is either using (SDK) or receiving (service) keys in lowercase - `sharedaccesskey=***` - and this just stops them from appearing in the test recordings. I can add comments to the function.